FABulous/FABulous_CLI/cmd_synthesis.py

```diff
@@ -0,0 +1,292 @@
+        clean
+
+    check:
+        hierarchy -check
+        stat
+
+    blif:
+        opt_clean -purge
+        write_blif -attr -cname -conn -param <file-name>
+
+    json:
+        write_json <file-name>
+"""
+
+
+synthesis_parser = Cmd2ArgumentParser(description=HELP)
+synthesis_parser.add_argument(
+    "files",
+    type=Path,
+    help="Path to the target files.",
+    completer=Cmd.path_complete,
+    nargs="+",
+)
+synthesis_parser.add_argument(
+    "-top",
+    type=str,
+    help="Use the specified module as the top module (default='top_wrapper').",
+    default="top_wrapper",
+)
+synthesis_parser.add_argument(
+    "-auto-top",
+    help="Automatically determine the top of the design hierarchy.",
+    action="store_true",
+)
+synthesis_parser.add_argument(
+    "-blif",
+    type=Path,
+    help="Write the design to the specified BLIF file. Writing of an output file is omitted "
+    "if this parameter is not specified.",
+    completer=Cmd.path_complete,
+)
+synthesis_parser.add_argument(
+    "-edif",
+    type=Path,
+    help="Write the design to the specified EDIF file. Writing of an output file is omitted "
+    "if this parameter is not specified.",
+    completer=Cmd.path_complete,
+)
+synthesis_parser.add_argument(
+    "-json",
+    type=Path,
+    help="Write the design to the specified JSON file. "
+    "If this parameter is not specified it will default to <first_file_stem>.json",
+    completer=Cmd.path_complete,
+)
+synthesis_parser.add_argument(
+    "-lut",
+    type=str,
+    default="4",
+    help="Perform synthesis for a k-LUT architecture (default 4).",
+)
+synthesis_parser.add_argument(
+    "-plib",
+    type=str,
+    help="Use the specified Verilog file as a primitive library.",
+    completer=Cmd.path_complete,
+)
+synthesis_parser.add_argument(
+    "-extra-plib",
+    type=Path,
+    help="Use the specified Verilog file for extra primitives (can be specified multiple times).",
+    action="append",
+    completer=Cmd.path_complete,
+)
+synthesis_parser.add_argument(
+    "-extra-map",
+    type=Path,
+    help="Use the specified Verilog file for extra techmap rules (can be specified multiple times).",
+    action="append",
+    completer=Cmd.path_complete,
+)
+synthesis_parser.add_argument(
+    "-encfile",
+    type=Path,
+    help="Passed to 'fsm_recode' via 'fsm'.",
+    completer=Cmd.path_complete,
+)
+synthesis_parser.add_argument(
+    "-nofsm",
+    help="Do not run FSM optimization.",
+    action="store_true",
+)
+synthesis_parser.add_argument(
+    "-noalumacc",
+    help="Do not run 'alumacc' pass. I.e., keep arithmetic operators in their direct form ($add, $sub, etc.).",
+    action="store_true",
+)
+synthesis_parser.add_argument(
+    "-carry",
+    type=str,
+    required=False,
+    choices=["none", "ha"],
+    default="none",
+    help="Carry mapping style (none, half-adders, ...) default=none.",
+)
+synthesis_parser.add_argument("-noregfile", help="Do not map register files.")
```

Comment:
```suggestion
synthesis_parser.add_argument(
    "-noregfile",
    help="Do not map register files.",
    action="store_true",
)
```
Sorry that one slipped through when I did the inital review.